cloudsecrets/secret.go

```diff
@@ -0,0 +1,10 @@
+package cloudsecrets
+
+import "context"
+
+type SecretStorage interface {
+	// FetchSecret
+	// Abstraction to fetch value from secret store
+	// GCP uses versioning for each secret, you can always fetch the latest version by setting versionId="latest"
+	FetchSecret(ctx context.Context, secretId string, versionId string) (string, error)
```

Comment:
Do we need secret versioning? I'm thinking that maybe we support `:latest` only.

And if we'd need versioning in the future we'd create additional interface.

Reply:
I had it like this before.. I don't really know what's the best approach. 

I have no idea why GCP is versioning the secrets in the first place. Anyway I'll do it as you suggest